middleware/validation.js

```diff
@@ -0,0 +1,14 @@
+import Exception from '../utils/exception.js';
+
+export const validate = (schema) => {
+  return (req, res, next) => {
+    const data = req.method === 'GET' ? req.query : req.body;
```

Comment:
all query parameters are strings
are you sure that the validation is performed correctly?